Framework/DataHandling/src/LoadILLIndirect2.cpp

```diff
@@ -273,6 +311,86 @@ void LoadILLIndirect2::loadDataIntoTheWorkSpace(NeXus::NXEntry &entry) {
   }
 }
 
+/**
+ * @brief LoadILLIndirect2::loadDiffractionData
+ * Load IN16B diffraction data from the Nexus file when requested.
+ * @param entry
+ */
+void LoadILLIndirect2::loadDiffractionData(NeXus::NXEntry &entry) {
+  NeXus::NXClass instrument = entry.openNXGroup("instrument");
+
+  // first, determine version
+  bool newVersion = instrument.containsDataSet("version");
+
+  // find the path to the data
+  std::string dataPath;
+  if (instrument.containsGroup("DiffDet")) {
+    dataPath = "instrument/DiffDet/data";
+  } else if (entry.containsGroup("dataDiffDet")) {
+    dataPath = "dataDiffDet/DiffDet_data";
+  } else {
+    throw std::runtime_error("Dataset path not found.");
+  }
+
+  NXData dataGroup = entry.openNXData(dataPath);
+
+  NXInt data = dataGroup.openIntData();
+  data.load();
+
+  NXData dataMonGroup = entry.openNXData("monitor/data");
+  NXInt dataMon = dataMonGroup.openIntData();
+  dataMon.load();
+
+  // First, Monitor
+  // Assign Y
+  int *monitor_p = &dataMon(0, 0);
+  m_localWorkspace->dataY(0).assign(monitor_p, monitor_p + m_numberOfChannels);
+  // Assign Error
+  MantidVec &E = m_localWorkspace->dataE(0);
+  std::transform(monitor_p, monitor_p + m_numberOfChannels, E.begin(),
+                 [](const double v) { return std::sqrt(v); });
+
+  if (!newVersion) {
+    // Then Tubes
+    PARALLEL_FOR_IF(Kernel::threadSafe(*m_localWorkspace))
+    for (int i = 0; i < static_cast<int>(m_numberOfTubes); ++i) {
+      for (size_t j = 0; j < m_numberOfPixelsPerTube; ++j) {
+        size_t index;
+        if (i == 2 || i == 3) {
+          index = (m_numberOfTubes - 1 - i) * m_numberOfPixelsPerTube +
+                  (m_numberOfPixelsPerTube - 1 - j) + m_numberOfMonitors;
+        } else {
+          index = (m_numberOfTubes - 1 - i) * m_numberOfPixelsPerTube + j +
+                  m_numberOfMonitors;
+        }
+        // Assign Y
```

Comment:
366-373 is repeated in both branches of the if clause. You can make a function that takes `*data_p` and `index` and does it, and call it twice correspondingly.

Reply:
I think the best solution is to move the if in the for and let the preprocessor do its job, so there is no repeats.

Reply:
That's another option, I still think if out of the for makes more sense.